src/api_server/src/http/mod.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use actix_web::{body::BoxBody, web, HttpRequest, HttpResponse};
+use jwt_simple::prelude::Ed25519PublicKey;
+use kbs_types::{Attestation, Challenge, ErrorInformation, Request};
+use std::sync::Arc;
+use strum_macros::EnumString;
+use tokio::sync::{Mutex, RwLock};
+
+use crate::attest::AttestVerifier;
+use crate::auth::validate_auth;
+use crate::resource::{set_secret_resource, Repository, ResourceDesc};
+use crate::session::{Session, SessionMap, KBS_SESSION_ID};
+use crate::token::TokenBroker;
+
+mod confidential_api;
+mod public_api;
+mod user_api;
+
+/// RESTful APIs that require Attestation and are protected by KBS Attestation Protocol
+pub use confidential_api::*;
+
+/// RESTful APIs that require user authentication
+pub use user_api::*;
+
+/// RESTful APIs that is public
+pub use public_api::*;
```

Comment:
I divide them into confidential, user, and public because calling some APIs of KBS need to go through the `R-C-A-R` four step attestation protocol, calling some APIs of KBS need to carry authentication tokens that represent the user's identity, and calling some APIs of KBS do not require any additional authentication. Therefore, organizing them according to this classification would make the code more maintainable. What do you think?

Reply:
I think splitting the API implementation into logical pieces makes a lot of sense and would help maintaining the code. And that made me think about how we'd want to split it, hence my proposal ;-)
And the distinction between RCAR related APIs and the rest of the API makes sense to me as well. But RCAR covers both `/auth` and `/attest`, which is what I tried to put under the `protocol.rs` file (although that one could be called `rcar.rs` as well).

The confidential/user/public split that you're proposing is based on credentials needed (although for example `/auth` is a public API that does not require authentication) rather than usage of the over all KBS interface. So I was proposing a usage based split instead: One part for the core KBS protocol (`{protocol|core|rcar}.rs`), that KBS users must go through if they want to be able to use the second part of the API (`attestation.rs`) in order to fetch any resource/item (secret or not) out of the KBS. 
Then we have part of our interface who is meant for administration/configuring the KBS itself (and that implicitly must be used by authenticated users), which is the `config.rs` part of my proposal.

Reply:
@sameo Good suggestion. Now we have three types of APIs: `core.rs`, `attestation.rs`, and `config.rs`. I think this is good and the division is clear. Thanks!

In addition, do we also need `public.rs` to organize public APIs that do not require attestation or user authentication? For example, getting cert chain.

Reply:
> @sameo Good suggestion. Now we have three types of APIs: `core.rs`, `attestation.rs`, and `config.rs`. I think this is good and the division is clear. Thanks!
> 

👍 

> In addition, do we also need `public.rs` to organize public APIs that do not require attestation or user authentication? For example, getting cert chain.

Yes, that makes sense to me.